src/transformers/generation_utils.py

```diff
@@ -747,6 +840,16 @@ def greedy_search(
 
             >>> print("Generated:", tokenizer.batch_decode(outputs, skip_special_tokens=True))
         """
+        logits = None
```

Comment:
I think a tuple would be nicer here. `torch.cat` reallocates new memory each time and since we don't know when generation stops we cannot reallocate all the new memory before. I'd prefer to have:

```python
logits = () if output_scores else None
decoder_attentions = () if output_attentions else None
decoder_hidden_states = () if output_hidden_states else None
```

Reply:
Adding a new tuple won't reallocate memory for the whole tensor and that's also how we handle it in the models with `output_attentions` and `output_hidden_states` :-)